src/main/java/ch/uzh/ifi/hase/soprafs21/rest/mapper/PathDTOMapper.java

```diff
@@ -9,14 +9,25 @@
 import org.mapstruct.Mapping;
 import org.mapstruct.factory.Mappers;
 
-@Mapper(uses = {SpatialDTOMapper.class})
+import java.util.List;
+
+@Mapper(uses = {SpatialDTOMapper.class,UserDTOMapper.class})
 public interface PathDTOMapper {
     PathDTOMapper INSTANCE = Mappers.getMapper(PathDTOMapper.class);
     SpatialDTOMapper SPATIAL_MAPPER = Mappers.getMapper(SpatialDTOMapper.class);
+    UserDTOMapper USER_DTO_MAPPER =Mappers.getMapper(UserDTOMapper.class);
 
     @Mapping(expression = "java(routeDto.getId())", target = "id")
```

Comment:
I have noticed that you map id from dto to entity when addPath. The id should not be mapped since it will be generated.

Reply:
The mappers should be as generic as possible, in the given context when we are calling the method to convert dto to entity it is pre verified that the id must be null so that it gets generated when the entity is persisted.

Reply:
In the api it is defined as readOnly attribute.  You can just put "target = "id", ignore = true"

Reply:
updated